packages/extension-api/src/extension-api.d.ts

```diff
@@ -788,6 +789,131 @@ declare module '@podman-desktop/api' {
     connection: ContainerProviderConnection;
   }
 
+  /**
+   * Callback for openning shell session
+   */
+  export interface ProviderConnectionShellAccess {
+    /**
+     * Opens new session using ProviderConnectionShellAccessImpl class
+     * @example
+     * ```typescript
+     * open(): ProviderConnectionShellAccessSession {
+     *  // Create stream 
+     *  #connected = true;
+     *  return {
+     *    onData: onData,
+     *    onError: onError,
+     *    onEnd: onEnd,
+     *    write: write.bind(this),
+     *    resize: resize.bind(this),
+     *    close: close,
+     *  };
+     * }
+     * ```
+     */
+    open(): ProviderConnectionShellAccessSession;
+  }
+
+  /**
+   * Callbacks for interaction with shell session
+   */
+  export interface ProviderConnectionShellAccessSession {
+    /**
+     * Receiving data event
+     * @example
+     * ```typescript
+     * onDataEmit = new EventEmitter<ProviderConnectionShellAccessData>();
+     * onData: Event<ProviderConnectionShellAccessData> = this.onDataEmit.event;
+     * 
+     * // later in code when "data" are emmited
+     * onDataEmit.fire({ data: data });
+     * ```
```

Comment:
I don't understand again, you're describing the code of Podman Desktop here

an extension will not fire anything to onData, just consume it

we need to use a code that extension will use/follow

so here, we don't describe what Podman Desktop core is doing, just how I can do things from my extension